app/Services/TotalBalanceCalculator.php

```diff
@@ -0,0 +1,26 @@
+<?php
+
+namespace App\Services;
+
+use App\Models\Entry;
+use Illuminate\Database\Eloquent\Collection;
+
+class TotalBalanceCalculator
+{
+    /**
+     * @param Collection|Entry[] $entries
+     * @return float
+     */
+    public function calculateBalance(iterable $entries): float
+    {
+        $total = 0;
+        foreach ($entries as $entry) {
+            // avoid floating point arithmetic
+            $value = (int)($entry->value * 100);
+            $total += $value;
+        }
+        $total = $total / 100;
```

Comment:
As you're only summing the values here, do you need to worry about floating point arithmetic? As you're passing in a Collection, could you have used the sum method? https://laravel.com/docs/8.x/collections#method-sum

```
$total = $entries->sum('value');
```